lib/util/viewer.js

```diff
@@ -47,20 +49,48 @@ export function routeIsValid (route, routeId) {
   return true
 }
 
+/**
+ * Run heuristic on pattern description to extract headsign from pattern description
+ * @param {*} pattern pattern to extract headsign out of
+ * @returns           headsign of pattern
+ */
+export function extractHeadsignFromPattern (pattern) {
+  let headsign = pattern.headsign
+  // In case stop time headsign is blank, extract headsign from the pattern 'desc' attribute
+  // (format: '49 to <Destination> (<destid>)[ from <Origin> (<originid)]').
+  if (isBlank(headsign)) {
+    const matches = pattern.desc.match(/(?: to )(.*?)(?: \()/)
+    if (matches) {
+      headsign = matches[1]
+    }
+  }
+
+  // If that regex didn't work, try the old regex
+  if (isBlank(headsign)) {
+    const matches = pattern.desc.match(/ to ([^(from)]+) \(.+\)/)
+    if (matches) {
+      headsign = matches[1]
+    }
+  }
+
+  // If the headsign is still blank, show the description
```

Comment:
Should we use the last stop of the pattern instead?

Reply:
This is a really cool idea and would work well, but we haven't yet grabbed stops from OTP when generating the headsigns.

Reply:
The reason is some agencies give non-descript names to the patterns, which in the end won't make sense to the user (try in the ATL config Cherokee County Routes 1 & 2, or the Gwinnett County "Sweeper" Bus).

Reply:
I agree. I think we will need to do that processing in OTP however, to avoid sending complete stop info for every single route.